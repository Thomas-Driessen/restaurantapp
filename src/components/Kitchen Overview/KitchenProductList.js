import React from 'react';
import Grid from '@material-ui/core/Grid';
import KitchenProduct from './KitchenProduct';
import Typography from '@material-ui/core/Typography';

class ProductsList extends React.Component {

    render(){
    return(
        <Grid item xs={12} sm={6} lg={4} xl={4} style={{padding: 25}}>
            <Typography variant="h4">{this.props.listTitle}</Typography>
            <div className="background-lightgray kitchen-overview-container">
                { this.props.products.map((currentProduct, index) => (
                    <KitchenProduct key={index} product={currentProduct} goToNext={this.props.goToNext}/>
                ))}
            </div>
        </Grid>
    )}
}
export default ProductsList