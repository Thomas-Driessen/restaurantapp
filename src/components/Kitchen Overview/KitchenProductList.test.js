import React from 'react';
import { render } from '@testing-library/react';
import KitchenProductList from './KitchenProductList';
import renderer from 'react-test-renderer';

const products = [
    {
        title: "Coca-Cola",
      },
      {
        title: "Pepsi",
      },
];

test('Render Product Coca-Cola', () => {
    const { getByText } = render(<KitchenProductList products={products}/>);
    const linkElement = getByText(/Coca-cola/i);
    expect(linkElement).toBeInTheDocument();
  });
  
  test('Render Product Pepsi', () => {
    const { getByText } = render(<KitchenProductList products={products}/>);
    const linkElement = getByText(/Pepsi/i);
    expect(linkElement).toBeInTheDocument();
  });

  it('KitchenProductList renders correctly', () => {
    const tree = renderer
      .create(<KitchenProductList products={products}/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
});