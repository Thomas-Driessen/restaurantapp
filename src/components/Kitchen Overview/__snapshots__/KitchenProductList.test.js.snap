// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KitchenProductList renders correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 MuiGrid-grid-xl-4"
  style={
    Object {
      "padding": 25,
    }
  }
>
  <h4
    className="MuiTypography-root MuiTypography-h4"
  />
  <div
    className="background-lightgray kitchen-overview-container"
  >
    <div>
      <div>
        <div
          className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-gutterBottom"
            >
              Coca-Cola
            </h2>
          </div>
          <div
            className="MuiCardActions-root MuiCardActions-spacing"
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-flex-end"
            >
              <div
                style={
                  Object {
                    "alignItems": "right",
                    "display": "flex",
                  }
                }
              >
                <button
                  aria-label="push"
                  className="MuiButtonBase-root MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div>
        <div
          className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-gutterBottom"
            >
              Pepsi
            </h2>
          </div>
          <div
            className="MuiCardActions-root MuiCardActions-spacing"
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-flex-end"
            >
              <div
                style={
                  Object {
                    "alignItems": "right",
                    "display": "flex",
                  }
                }
              >
                <button
                  aria-label="push"
                  className="MuiButtonBase-root MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
